### PR TITLE
Add an ability to update timezone

### PR DESCRIPTION
**Pull request by fastiuk**

### CODE REVIEW

**Comment by fastiuk** [comment]
@Meir-renford, here is a host-service code

File: scripts/hostcfgd [hunk 1]
```diff
@@ -1405,6 +1405,20 @@ class DeviceMetaCfg(object):
...
             run_cmd(['sudo', 'monit', 'reload'])
 
+    def timezone_update(self, data):
+        """
+        Apply timezone handler.
+        Run the following command in Linux: timedatectl set-timezone <timezone>
+        Args:
+            data: Read table's key's data.
+        """
+        new_timezone = data.get('timezone')
```
Comment (dgsudharsan) [inline scripts/hostcfgd:1419]: Should have cache and compare against existing value to avoid calling this command for any update in device_metadata?

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: scripts/hostcfgd (hunk 1 above)
Comment (fastiuk) [reply scripts/hostcfgd:1419]: Make sense, will update

File: scripts/hostcfgd [hunk 2]
```diff
@@ -1405,6 +1409,35 @@ class DeviceMetaCfg(object):
...
+        syslog.syslog(syslog.LOG_DEBUG,
+                      f'DeviceMetaCfg: timezone update to {new_timezone}')
+
+        if new_timezone is None:
+            syslog.syslog(syslog.LOG_ERR,
```
Comment (dgsudharsan) [inline scripts/hostcfgd]: This check mandates timezone config to be present in DEVICE_METADATA, else would throw error. I don't think its correct. It should be optional field.

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

**Review by dgsudharsan — APPROVED** [review]

**Pull request closed without merging**